src/unstract/adapters/llm/ollama/src/ollama.py

```diff
@@ -0,0 +1,94 @@
+import logging
+import os
+from typing import Any
+
+from httpx import ConnectError, HTTPStatusError
+from llama_index.core.llms import LLM
+from llama_index.llms.ollama import Ollama
+
+from unstract.adapters.exceptions import AdapterError
+from unstract.adapters.llm.constants import LLMKeys
+from unstract.adapters.llm.helper import LLMHelper
+from unstract.adapters.llm.llm_adapter import LLMAdapter
+
+logger = logging.getLogger(__name__)
+
+
+class Constants:
+    MODEL = "model"
+    API_KEY = "api_key"
+    TIMEOUT = "timeout"
+    BASE_URL = "base_url"
+    JSON_MODE = "json_mode"
+    CONTEXT_WINDOW = "context_window"
+
+
+class OllamaLLM(LLMAdapter):
+    def __init__(self, settings: dict[str, Any]):
+        super().__init__("Ollama")
+        self.config = settings
+
+    @staticmethod
+    def get_id() -> str:
+        return "ollama|4b8bd31a-ce42-48d4-9d69-f29c12e0f276"
+
+    @staticmethod
+    def get_name() -> str:
+        return "Ollama"
+
+    @staticmethod
+    def get_description() -> str:
+        return "Ollama AI LLM"
+
+    @staticmethod
+    def get_icon() -> str:
+        return "/icons/adapter-icons/ollama.png"
+
+    @staticmethod
+    def get_json_schema() -> str:
+        f = open(f"{os.path.dirname(__file__)}/static/json_schema.json")
+        schema = f.read()
+        f.close()
+        return schema
+
+    def get_llm_instance(self) -> LLM:
+        try:
+            llm: LLM = Ollama(
```

Comment:
Please check if `Ollama` has max_retries as an arg -> include that as well and add it to JSON schema if that's the case

Reply:
@chandrasekharan-zipstack Yes, checked this. Ollama doesnt have support for `max_retries.` Rather they support timeout. Have added that. 
Ref - https://docs.llamaindex.ai/en/stable/api_reference/llms/ollama/